cdp/smart_contract.py

```diff
@@ -0,0 +1,324 @@
+
+        def __repr__(self) -> str:
+            """Return a string representation of the Type."""
+            return str(self)
+
+    class TokenOptions:
+        """Options for ERC20 token contracts."""
+
+        def __init__(self, name: str, symbol: str, total_supply: int | Decimal | str):
+            self.name = name
+            self.symbol = symbol
+            self.total_supply = total_supply
+
+    class NFTOptions:
+        """Options for ERC721 NFT contracts."""
+
+        def __init__(self, name: str, symbol: str, base_uri: str):
+            self.name = name
+            self.symbol = symbol
+            self.base_uri = base_uri
+
+    class MultiTokenOptions:
+        """Options for ERC1155 multi-token contracts."""
+
+        def __init__(self, uri: str):
+            self.uri = uri
+
+    def __init__(self, model: SmartContractModel) -> None:
+        """Initialize the SmartContract class.
+
+        Args:
+            model (SmartContractModel): The model representing the smart contract.
+
+        Raises:
+            ValueError: If the smart contract model is empty.
+
+        """
+        if not model:
+            raise ValueError("SmartContract model cannot be empty")
+        self._model = model
+        self._transaction = None
+
+    @property
+    def smart_contract_id(self) -> str:
+        """Get the smart contract ID.
+
+        Returns:
+            str: The smart contract ID.
+
+        """
+        return self._model.smart_contract_id
+
+    @property
+    def network_id(self) -> str:
+        """Get the network ID of the smart contract.
+
+        Returns:
+            str: The network ID.
+
+        """
+        return self._model.network_id
+
+    @property
+    def wallet_id(self) -> str:
+        """Get the wallet ID that deployed the smart contract.
+
+        Returns:
+            str: The wallet ID.
+
+        """
+        return self._model.wallet_id
+
+    @property
+    def contract_address(self) -> str:
+        """Get the contract address of the smart contract.
+
+        Returns:
+            str: The contract address.
+
+        """
+        return self._model.contract_address
+
+    @property
+    def deployer_address(self) -> str:
+        """Get the deployer address of the smart contract.
+
+        Returns:
+            str: The deployer address.
+
+        """
+        return self._model.deployer_address
+
+    @property
+    def type(self) -> str:
+        """Get the type of the smart contract.
+
+        Returns:
+            str: The smart contract type.
+
+        Raises:
+            ValueError: If the smart contract type is unknown.
+
+        """
+        if self._model.type == SmartContractTypeModel.ERC20:
+            return self.Type.ERC20
+        elif self._model.type == SmartContractTypeModel.ERC721:
+            return self.Type.ERC721
+        elif self._model.type == SmartContractTypeModel.ERC1155:
+            return self.Type.ERC1155
+        else:
```

Comment:
This can use the parent string constructor once the `Type` class inherits from `Enum`